src/test/resources/log4j-custom.properties

```diff
@@ -0,0 +1,5 @@
+# Set everything to be logged to the console
+log4j.rootCategory=INFO, console
+log4j.appender.console=org.apache.log4j.ConsoleAppender
+log4j.appender.console.layout=org.apache.log4j.PatternLayout
+log4j.appender.console.layout.ConversionPattern=%d{yy/MM/dd HH:mm:ss} %p %c{1}: %m%n
```

Comment:
We must pass a custom log4j file if we want to see application logs when using serverless spark. The current `start.sh` passes one that spark actually fails to load, which triggers it to load its package default config. This causes spark to be able to log from its packages, but applicatoin logs (com.google.etc..) are ignored.

This file is used by the `start-general.sh`, it lives in the test folder, so is not bundled with the jar. 
The script basically does the following:
```
gsutil cp log4j-custom.properties "gs://yadavaja-sandbox-tsell-us/config/log4j-custom.properties"
```
Then in the submit spark command it has:
```
  --properties=spark.driver.extraJavaOptions='-Dlog4j.debug=true -Dlog4j.configuration=file:log4j-custom.properties' \
  --files=gs://yadavaja-sandbox-tsell-us/config/log4j-custom.properties \
```
The `--files` flag downloads it to all locations in the working dir and the `log4j.configuration=file:log4j-custom.properties` property tells it to load from the working dir.

If you also add a debug flag to extraJavaOptions `-Dlog4j.debug=true`, then at the start of the logs you will see:
```
log4j: Using URL [file:log4j-custom.properties] for automatic log4j configuration.
log4j: Reading configuration from URL file:log4j-custom.properties
... etc
log4j: Finished configuring.
```

After that you can now see the application logs, eg:
```
21/12/02 08:03:46 INFO GeneralTemplate: Parsing arguments [--config, config.yaml]
21/12/02 08:03:46 INFO GeneralTemplate: Loading config from file:///tmp/srvls-batch-f4c0cd26-2e2e-4fdb-841d-c021d0fb1e04/config.yaml
21/12/02 08:03:46 INFO GeneralTemplate: File fetched
21/12/02 08:03:46 INFO GeneralTemplate: Parsing yaml
21/12/02 08:03:47 INFO GeneralTemplate: Config loaded
GeneralTemplateConfig{input={shakespeare=InputConfig{format=bigquery, options={table=bigquery-public-data.samples.shakespeare}, path=null}}, output={wordspercorpus=OutputConfig{format=avro, path=gs://yadavaja-sandbox-tsell-us/output/wordspercorpus/, options={}, mode=Overwrite}}, query={wordspercorpus=QueryConfig{sql=SELECT corpus, count(*) distictwords, sum(word_count) numwords FROM shakespeare GROUP by corpus ORDER BY numwords DESC
}}}
```

Reply:
@snrssc thoughts on this?